conda/cli/main_search.py

```diff
@@ -141,7 +139,7 @@ def execute_search(args, parser):
     ms = None
     if args.regex:
         if args.spec:
-            ms = ' '.join(args.regex.split('='))
+            ms = MatchSpec(arg2spec(args.regex))
```

Comment:
Using `arg2spec` is critical for consistency. This is a bug fix we'd need to do even if this particular PR didn't do it.